sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -123,4 +159,55 @@ export class WorkloadIdentityCredential implements TokenCredential {
     }
     return this.azureFederatedTokenFileContent;
   }
+
+  private async requestOidcToken(
+    oidcRequestUrl: string,
+    systemAccessToken: string,
+  ): Promise<string> {
+    console.log("Requesting OIDC token from Azure DevOps...");
+    console.debug(oidcRequestUrl);
+
+    const requestOptions = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${systemAccessToken}`,
+      },
+    };
+
+    const response = await fetch(oidcRequestUrl, requestOptions);
+    const result = await response.json();
+    return result;
+  }
+
+  private checkDevopsSystemVars(): boolean {
+    if (
+      process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI &&
+      process.env.SYSTEM_TEAMPROJECTID &&
+      process.env.SYSTEM_PLANID &&
+      process.env.SYSTEM_JOBID &&
+      process.env.SECRET_SYSTEM_ACCESSTOKEN
+    ) {
+      return true;
+    }
+    if (!process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI) {
+      this.systemError = "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI,";
```

Comment:
I see that we're building this systemError and then setting it as an instance variable - where do we actually use it / throw it? 

I also wonder if it would be cleaner to collect missing env var names in an array and then creating the error if the array is not empty (pseudocode below)

```ts
const missingEnvVars = []
if (!process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI) missingEnvVars.push("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI")
...
...
if (missingEnvVars.length > 0) this.systemError = "Missing system variable(s) - ${missingEnvVars.join(', ')}"
```

Last question here is: is this a developer error or an SDK error for these to be missing? They're set by Azure DevOps right?

Reply:
They are set by Azure Devops. So definitely not an error by developer. but by Azure Devops

Reply:
done